src/otx/core/model/module/instance_segmentation.py

```diff
@@ -169,11 +169,13 @@ def _convert_pred_entity_to_compute_metric(
             inputs.polygons,
             inputs.labels,
         ):
-            rles = (
-                [encode_rle(mask) for mask in masks.data]
-                if len(masks)
-                else polygon_to_rle(polygons, *imgs_info.ori_shape)
-            )
+            rles = []
+            if len(bboxes):
```

Comment:
Why is this check needed?

Reply:
To prevent unexpected errors, inputs from val set should be checked for images that lack annotations.

Reply:
What do you exactly mean? What is the exact requirement for the training step? Is `len(bboxes) == len(rle)`? If the malformed input is given, we should stop and raise the exception, not rolling the wheel by introducing if-branching for the exceptional case.

Reply:
No, it has nothing to do with malformed or rolling the wheel. It's exactly as I mentioned: some images in the validation set lack annotations because we don't apply the same filtering as we do during training. If you believe we should include pre-filtering for val set, we can include it in the pre-filtering process (`src/otx/core/data/pre_filtering.py`). If not, we should anticipate that inputs.bboxes will yield bboxes = [], therefore bypassing RLE postprocessing.

Reply:
I read your comment twice but I cannot still understand it. Is there any case that `bboxes = []` but `len(masks) > 0` or `len(polygons)`?. To me, `bboxes` and `masks or polygons` are independent entities each other. However, you injected a constraint that `bboxes == [] => rle = []` From this statement and my conjecture, I asked you whether there is a strict constraint that `len(bboxes) == len(masks)` (`len(polygons)`).

Reply:
The requirement of `len(bboxes) == len(masks)` or `len(bboxes) == len(polygons)` is necessary. And, I understand how it might lead to confusion. So I've decided to remove this check and manage it within the polygon to RLE conversion function.